### ISSUE TITLE
Using as Library

### ISSUE THREAD

**Issue by teqdruid** [tidwall/tile38#18]
I'm thinking of using tile38 as a library rather than separate process. Looks like I can just instantiate a controller/collection.Collection and go. Is that the best drop-in place? I'd basically like access to most of your features, but I don't need/want replication and I can't afford a separate daemon and the complexity that entails.

Also, with regard to my tweet: when wrapping around -180/180, at what point in the code does mapping to multiple geometries occur?

**Comment by tidwall** [comment]
Correct. The `Collection.New()` is the way to go. A collection expects input from the geojson package, but you can generate a very lightweight `geojson.Object` without actual geojson by doing something like:

```go
col := collection.New()
point := geojson.New2DPoint(lon, lat)
col.ReplaceOrInsert("mypoint", point, nil, nil)
```

Then you can use the `col.Search()` function for retrieving the objects.

I don't publish documentation on methods under the collection package and I can't promise that these APIs won't change in the future. I recommend that you fork or vendor the repo.

You can find the wrapping around stuff in the [norm.go](https://github.com/tidwall/tile38/blob/1.2.0/index/norm.go) file. The [index.Search](https://github.com/tidwall/tile38/blob/1.2.0/index/index.go#L151) function does the mapping of the split rects.

Basically the `normRect()` and `normPoint()`functions are responsible for making sure that a point and a rectangle have sane positions within -90/+90,-180/+180. A rectangle might be split into multiples for overflows and wrapping.

I hope this helps and let me know if you have any more questions.

**Comment by teqdruid** [comment]
Cool!

Yeah, looks to me like you're structured somewhat for this, but some of the functionality is mixed into the server aspects a bit. The collection interface is also clearly designed for the server functionality rather than local clients. I was thinking I'd fork it and make modifications as necessary and time allows.

**Comment by tidwall** [comment]
Everything under the controller package is intended for server use. But you should be able to import the collection package and use it directly.
